### PR TITLE
vision in sim

### PR DESCRIPTION
**Pull request by lmaxwell24**
implements #31

### CODE REVIEW

File: subsystems/visionsubsystem.py [hunk 1]
```diff
@@ -101,10 +109,143 @@ def periodic(self) -> None:
...
+        self.camToTargYaw = Rotation2d(self.camToTarg.X(), self.camToTarg.Y())
+        self.camToTargPitch = Rotation2d(self.camToTargDistXY, -self.camToTarg.Z())
+        self.camToTargAngle = Rotation2d(
+            hypot(self.camToTargYaw.radians(), self.camToTargPitch.radians())
+        )
+
+        self.targToCam = Transform3d(targetPose, cameraPose)
+        self.targToCamYaw = Rotation2d(self.targToCam.X(), self.targToCam.Y())
+        self.targToCamPitch = Rotation2d(self.camToTargDistXY, -self.targToCam.Z())
+        self.targToCamAngle = Rotation2d(
```
Comment (icai0) [inline subsystems/visionsubsystem.py:142]: what angle is this

File: subsystems/visionsubsystem.py [hunk 2]
```diff
@@ -101,10 +109,143 @@ def periodic(self) -> None:
...
+
+            self.poseList.append(
+                EstimatedPose(botPose, seeTag, Timer.getFPGATimestamp())
+            )
+
+
+class RNG:
+    def __init__(self, stdDev: float, number: int) -> None:
+        self.stdDev = stdDev
+        self.number = number
```
Comment (icai0) [inline subsystems/visionsubsystem.py]: not used anywhere

File: subsystems/visionsubsystem.py [hunk 3]
```diff
@@ -101,10 +109,143 @@ def periodic(self) -> None:
...
+        # self.rngIdx = 0
+
+    def getNormalRandom(self) -> float:
+        return sin(1000000 * Timer.getFPGATimestamp()) * self.stdDev
+        # self.rngIdx = (self.rngIdx + 1) % self.number
+        # return self.rng[self.rngIdx]
+
+
+class VisionSubsystem(Subsystem):
+    # this is really bad
```
Comment (icai0) [inline subsystems/visionsubsystem.py:237]: :(

**Review by icai0 — CHANGES_REQUESTED** [review]

File: subsystems/visionsubsystem.py [hunk 4]
```diff
@@ -101,10 +109,143 @@ def periodic(self) -> None:
...
+                    botPose = Pose3d(
+                        simPose3d.X() + self.rng.getNormalRandom(),
+                        simPose3d.Y() + self.rng.getNormalRandom(),
+                        simPose3d.Z() + self.rng.getNormalRandom(),
+                        simPose3d.rotation(),
+                    )
+
+            rel = CameraTargetRelation(simPose3d + camera.location, botPose)
+            VisionSubsystemReal.updateAdvantagescopePose(
+                botPose, camera.key, simPose3d, rel.camToTarg
```
Comment (icai0) [inline subsystems/visionsubsystem.py]: first argument should be the camera's pose

Comment (icai0) [reply subsystems/visionsubsystem.py]: image
axis on the ground comes from botPose

**Review by icai0 — APPROVED** [review]

**Review by LandonBayer — APPROVED** [review]

**Pull request merged**